src/survey/controllers/create_survey_controller.py

```diff
@@ -1,49 +1,75 @@
-        completed_match_id = self._data.get("completed_match_id")
-        if explanation is None:
-            return failure_response(
-                "POST body is misformatted", status.HTTP_400_BAD_REQUEST
-            )
+        if did_meet is None:
+            return failure_response("did_meet", status.HTTP_400_BAD_REQUEST)
 
-        # Get optional fields
+        # Get optional fields based on did_meet
+        did_meet_reason = self._data.get("did_meet_reason")
+        did_not_meet_reasons = self._data.get("did_not_meet_reasons")
         rating = self._data.get("rating")
-        if rating is not None and rating not in RATINGS:
-            return failure_response(
-                "The provided rating is invalid", status.HTTP_400_BAD_REQUEST
+        if did_meet:
+            if rating is None:
+                return failure_response(
+                    "Rating is required for a completed match",
+                    status.HTTP_400_BAD_REQUEST,
+                )
+            elif rating not in constants.RATINGS:
+                return failure_response(
+                    "The provided rating is invalid", status.HTTP_400_BAD_REQUEST
+                )
+        else:
+            if not did_not_meet_reasons:
+                return failure_response(
+                    "did_not_meet_reasons is required if did_meet is False",
+                    status.HTTP_400_BAD_REQUEST,
+                )
+            if not all(
+                map(lambda x: x in constants.DID_NOT_MEET_LONG, did_not_meet_reasons)
+            ):
+                return failure_response(
+                    "did_not_meet_reasons has invalid elements",
+                    status.HTTP_400_BAD_REQUEST,
+                )
+
+            did_not_meet_reasons = list(
+                map(lambda x: constants.long_to_short(x), did_not_meet_reasons)
             )
+            if len(did_not_meet_reasons) > 5:
+                return failure_response(
+                    "Too many elements in did_not_meet_reasons",
+                    status.HTTP_400_BAD_REQUEST,
+                )
 
         # Verify that required ids are valid
-        completed_match = Match.objects.filter(id=completed_match_id)
+        completed_match = Match.objects.filter(id=self._match_id)
         if not completed_match:
-            return failure_response(
-                "completed_match_id is invalid", status.HTTP_404_NOT_FOUND
-            )
+            return failure_response("match_id is invalid", status.HTTP_404_NOT_FOUND)
         completed_match = completed_match[0]
 
         # Check if the submitting user has already submitted a survey
-        submitting_user = Person.objects.filter(id=self._request.user.id)[0]
+        submitting_person = self._request.user.person
         survey = Survey.objects.filter(
-            submitting_user=submitting_user, completed_match=completed_match
+            submitting_person=submitting_person, completed_match=completed_match
```

Comment:
this looks a bit convoluted

this might be better: `Survey.objects.filter(submitting_person=self._request.user.person)`

remember that you can always do `Survey.objects.filter(submitting_user__person=self._request.user.person)` but here it's not necessary to do that bc you guys just did a weird naming thing